homeassistant/components/tplink/__init__.py

```diff
@@ -76,6 +77,10 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up TPLink from a config entry."""
     config_data = hass.data[DOMAIN].get(ATTR_CONFIG)
 
+    device_registry = dr.async_get(hass)
+    tplink_devices = dr.async_entries_for_config_entry(device_registry, entry.entry_id)
```

Comment:
The name indicates that this is a coro, should it be awaited or is this also one of those magical non-coro methods with the `async_` prefix? :-)

Reply:
The latter: https://github.com/home-assistant/core/blob/master/homeassistant/helpers/device_registry.py#L624

(async_get() replaces the deprecated async_get_registry(). It is/should be called from async context but it's not a coroutine)

Reply:
okay, the naming scheme is really confusing if you are not digging deeper into the code while reviewing  (not your fault!), so I just wanted to be sure about it :+1: